src/bootstrap/src/core/config/config.rs

```diff
@@ -1314,7 +1314,13 @@ impl Config {
 
         // Set flags.
         config.paths = std::mem::take(&mut flags.paths);
-        config.skip = flags.skip.into_iter().chain(flags.exclude).collect();
+        config.skip = flags
+            .skip
+            .into_iter()
+            .chain(flags.exclude)
+            .map(|p| t!(config.src.join(p).canonicalize()))
```

Comment:
On a native `x86_64-pc-windows-msvc` machine (not in msys2), I tried:

Before this PR:

```
PS X:\repos\rust> ./x test .\tests\ui\compiletest-self-test --skip .\tests\ui\compiletest --stage 1 --force-rerun
Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.05s
Building stage0 library artifacts (x86_64-pc-windows-msvc)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.43s
Building compiler artifacts (stage0 -> stage1, x86_64-pc-windows-msvc)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.56s
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
Building stage1 library artifacts (x86_64-pc-windows-msvc)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.38s
Building stage0 tool compiletest (x86_64-pc-windows-msvc)
    Finished `release` profile [optimized + debuginfo] target(s) in 0.46s
Testing stage1 compiletest suite=ui mode=ui (x86_64-pc-windows-msvc)

running 6 tests
......
```

After this PR:

```
PS X:\repos\rust> ./x test .\tests\ui\compiletest-self-test --skip .\tests\ui\compiletest --stage 1 --force-rerun
Building bootstrap
   Compiling bootstrap v0.0.0 (X:\repos\rust\src\bootstrap)
    Finished `dev` profile [unoptimized] target(s) in 7.91s
thread 'main' panicked at src\bootstrap\src\core\config\config.rs:1321:22:
config.src.join(p).canonicalize() failed with The system cannot find the file specified. (os error 2)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Build completed unsuccessfully in 0:00:08
```

Looks like it's working as intended, though the error/panic message isn't particularly helpful. Should we specify that it's the `--skip` validation that failed and which path?

Reply:
or just use absolute() instead of canonicalize? there's already error handling later for paths that don't exist i think.

Reply:
> or just use absolute() instead of canonicalize? there's already error handling later for paths that don't exist i think.

There is no error-handling on --skip and --exclude flags for the paths that don't exist.